app/models/spree/item_adjustments.rb

```diff
@@ -12,7 +12,7 @@ def initialize(item)
     end
 
     def update
-      update_adjustments if item.persisted?
+      update_adjustments if updatable_totals?
```

Comment:
Is this a sign that callbacks are more complicated than calling it manually?

Reply:
So this new class (`ItemAdjustments`) is being introduced here, but I've only started using it with shipments. The idea is that it will also be used with other things like line_items and payments etc but there's some additional changes needed before that can happen and the class can be used more widely.  I've left that out here as it would make the PR substantially larger. 

But yeah, the idea is that it will get used in certain callbacks.

Reply:
:fire: callbacks! :hammer: :wink: